File: diltakip/src/main/java/com/dilokul/diltakip/model/Teacher.java
package com.dilokul.diltakip.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Teacher {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", columnDefinition = "CHAR(36)")
    private UUID id;

    private String name;
    private String tckn;
    private String surname;
    private String phone;
    private String secondPhone;
    private String address;

    @UpdateTimestamp
    @Column(name = "update_time")
    private LocalDateTime updateTime;

    @Column(name = "created_on", nullable = false, updatable = false)
    @CreationTimestamp
    private LocalDateTime createdOn;

    @OneToMany(mappedBy = "teacher", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Class> classes;
}

File: diltakip-fe/src/interface/Teacher.ts
// Teacher.ts
export interface Teacher {
    id: number;
    name: string;
    tckn: string;
    surname: string;
    phone: string;
    city: string;
    secondPhone?: string;
    address: string;
    registrationDate: string;
    updateTime?: string;
    createdOn: string;
}

export interface Page<T> {
    content: T[];
    totalPages: number;
    totalElements: number;
    size: number;
    number: number;
}